zerver/actions/message_delete.py

```diff
@@ -60,12 +60,12 @@ def do_delete_messages(realm: Realm, messages: Iterable[Message]) -> None:
 
     sample_message = messages[0]
     message_type = "stream"
-    users_to_notify = []
+    users_to_notify = set()
     if not sample_message.is_stream_message():
         assert len(messages) == 1
         message_type = "private"
         ums = UserMessage.objects.filter(message_id__in=message_ids)
-        users_to_notify = [um.user_profile_id for um in ums]
+        users_to_notify = set(ums.values_list("user_profile_id", flat=True))
```

Comment:
This is also a nice performance improvement to not query full Django objects that we're not going to process.